euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceByKey.java

```diff
@@ -352,19 +391,58 @@ public boolean isCombinable() {
     return DAG.of(reduceState);
   }
 
+  static <VALUE> ReduceFunctor<VALUE, VALUE> toReduceFunctor(
+      CombinableReduceFunction<VALUE> reducer1) {
+
+    return new ReduceFunctor<VALUE, VALUE>() {
+      @Override
+      public boolean isCombinable() {
+        return true;
+      }
+
+      @Override
+      public void apply(Iterable<VALUE> elem, Context<VALUE> context) {
+        context.collect(reducer1.apply(elem));
+      }
+    };
+  }
+
+
+  public static class SingleValueContext<T> implements Context<T> {
+
+    T value;
+
+    @Override
+    public void collect(T elem) {
+      value = elem;
+    }
+
+    @Override
+    public Object getWindow() {
+      throw new UnsupportedOperationException(
+          "The window is unknown when applying combinable function");
```

Comment:
* i don't think this is generally true. i suppose the difficulty here is "merely" getting at the corresponding context information during the reduction through the state. and it indicates to me that the (api) boundary between the operator and the state implementation is currently insufficient. (this memory leak appears to be repeated also in the spark and flink RBK translators.)
* further, i don't think exposing this `SingleValueContext` is a good idea (it seems to me like an impl detail of `CombiningReduceState`).